docs/conceptual/cryptography/keys.md

```diff
@@ -0,0 +1,145 @@
+---
+title: Substrate Keys
+---
+
+Substrate uses multiple sets of public/private key pairs to represent
+participants of the network.
+
+## Overview
+
+Blockchain systems involve a number of participants playing different roles in
+the system. The Substrate node uses a nominated proof of stake (NPoS) system by
+default. In this system, there are two core roles:
+
+1. Validators: Users that produce blocks.
+2. Nominators: Users that support a validator.
+
+Validators need to collect funds which back their intention to produce blocks.
+Validators can provide their own funds, and also collect funds from nominators
+who trust the validator to produce new blocks correctly. As a reward for doing
+things correctly, both validators and their respective nominators get paid.
+These funds can be used for slashing if the validator misbehaves.
+
+> **Note:** Learn more about validators and nominators in the context of the
+> Substrate's NPoS [Staking module](/rustdocs/master/srml_staking/index.html).
+
+Both validators and nominators may be working with a significant value of funds,
+so Substrate has introduced abstractions which allow safer behaviors for end
+users participating in the network.
+
+These abstractions are:
+
+* [Account Keys](#account-keys): Accounts that should hold funds.
+    * [Stash Keys](#stash-keys): Cold wallets controlled by controller keys.
+    * [Controller Keys](#controller-keys): Semi-online keys that will be in the
+      direct control of a user and used to submit extrinsics.
+* [Session Keys](#session-keys): Hot keys that be must kept online by a
+  validator to perform network operations and should not hold funds.
+
+## Account Keys
+
+Account keys represent the normal kind of accounts you might expect from other
+blockchain systems. These accounts hold funds, and in the context of Substrate's
+Balances module, must have at least an existential deposit amount of those
+funds. Accounts keys are defined generically and its type is ultimately made
+concrete in the runtime. They interact with the blockchain through transactions.
+
+In the context of Substrate's Staking module, there are two abstractions on top
+of account keys:
+
+1. Stash Keys
+2. Controller Keys
+
+These keys are distinguished by their intended use, not by any underlying
+cryptographic difference. When creating new controller or stash keys, all
+cryptography supported by account keys are an available option.
+
+### Stash Keys
+
+Stash keys are intended to behave like a cold wallet, existing on a piece of
```

Comment:
Should not mention voting here because you can actually designate a separate Proxy account. Controller can't vote unless you designate the same account as Proxy and Controller.

--------------------------

The Stash keys are the public/private key pair that defines a Stash account. This account is like a "savings account" in that you should not make frequent transactions from it. Therefore, its private key should be treated with the utmost security, for example protected in a safe or layers of hardware security.

Since the Stash key is kept offline, it designates a Controller account to make non-spending decisions with the weight of the Stash account's funds. It can also designate a Proxy account to vote in governance [link] on its behalf.